devDocs/buildSystemNotes.md

```diff
@@ -0,0 +1,23 @@
+A Python virtual environment is now used transparently by the NVDA build system, and all Python dependencies are installed into this environment using `pip`.
+
+From a developer's perspective using the NVDA build system, there is no need to worry about the Python virtual environment behind the scenes, nor should you create and or activate one manually. All NVDA build system commands will handle this transparently.
+
+* To build NVDA, SCons should continue to be used in the usual way. E.g. executing scons.bat in the root of the repository. Running `py -m SCons`  is no longer supported, and scons.py has also been removed. 

```

Comment:
I don't think we need to go over how to build, run, or execute tests here. Those should be documented in a single place, currently that is the main readme. I'd make this doc more of a how / why explanation.

I'd certainly references to unsupported "old ways" of doing this. The change log / PR can used to see what has changed. The python file is no longer available, so it would be pretty hard to make that mistake.

I think we should add an explicit line that builds / running should be done from a cmd.exe window, but that probably belongs in the main readme.md file.

After the first little introduction about how "the build system uses virtual environments transparently", I'd just have a couple of headings.
- How the build system works
  This should highlight the names of the main entry point scripts, list all scripts with an overview of their purpose, and certainly point out they all rely on `ensureVenvAndActivate`.
- Motivation for using virtual environments
  Eg In order to ensure the build environment is clean, no conflicts with other installed packages

Reply:
Perhaps you can have a go at re-wording / re-structuring this doc if you have some good ideas in your head already. Though I have taken all your other suggestions.

Reply:
Yep, I'll take a look at this.